docs/DeveloperGuide.md

```diff
@@ -1,38 +1,536 @@
+
+![commandMarkDiffClass.png](UML%2FCommands%2FcommandMarkDiffClass.png)
+The CommandMarkDifficulty class includes the following key components:
+- `qnIndex`: An integer representing the question number to be marked for difficulty.
+- `nDifficulty`: An enumeration representing the difficulty level to be assigned to the question.
+
+#### Implementation of Markdiff Command
+Developers can use the `CommandMarkDifficulty` class as a template for handling difficulty marking commands in the 
+quizhub application. Here are the key steps for implementing this class:
+
+- **Parsing User Input**: Parse the user input to extract the question number and the specified difficulty level.
+
+- **Validation**: Implement validation logic to ensure that the user input is correctly formatted and contains valid 
+information.
+
+- **Marking Difficulty**: Implement the logic to execute the operation of marking the question with the specified 
+difficulty level. This typically involves invoking methods in the QuestionList and Question classes to update the 
+uestion's difficulty.
+
+- **Data Persistence**: If necessary, update the data storage to save the changes. In the provided code, the 
+`dataStorage.updateData(questions)` method is used to save changes to the question list.
+
+- **Error Handling**: Handle any exceptions or errors that may occur during the marking operation and provide 
+appropriate feedback to the user.
+
+
+### Command Exit - Exit Program
+
+#### Brief Description of Command Exit
+
+Command Exit is responsible for exiting the program
+- `bye` - to exit program
+
+
+#### Class structure of Command Exit
+
+![commandExitSequence.png](UML%2FCommands%2FcommandExitSequence.png)
+
+When the user initiate Command Exit, any unsaved data will be saved into storage and thereafter, the exit message will
+be displayed.
+
+## Storage Component
+
+The Storage class is a critical component of the QuizHub application, responsible for managing the storage, retrieval, and updating of question data. It represents a hard disk storage system for storing, reading, and updating question data. It interacts with question data in a file specified by the user and integrates with the `QuestionList` class for effective storage and retrieval of questions.
+
+Developers need to be aware of the two main methods for loading and updating data:
+
+- `public void loadData(QuestionList questions)`
+- `public void updateData(QuestionList questions)`
+
+The details of the data loading and updating process are explained in details below.
+
+### Constructor
+
+#### `public Storage(String filePath)`
+
+- **Parameters:**
+  - `filePath`: The path to the file where question data is stored and read from.
+
+### Loading data
+
+The process of loading data from the storage file specified in the constructor takes places in a few steps. To illustrate the overall flow on loading data, refer to the sequence diagram below.
+
+![](./UML/Storage.jpg)
```

Comment:
Is it ok for the methods to create the nested activation bar in Storage like this?